File: src/main/java/pl/com/tegess/config/security/AdminAuthentication.java
package pl.com.tegess.config.security;

import org.springframework.security.core.Authentication;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;
import pl.com.tegess.domain.admin.Admin;

import java.util.Collection;

public class AdminAuthentication implements Authentication {

    private final Admin admin;
    private boolean authenticated;


    public AdminAuthentication(Admin admin) {
        this.admin = admin;
        this.authenticated = true;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return admin.getAuthorities();
    }

    @Override
    public Object getCredentials() {
        return admin.getPassword();
    }

    @Override
    public Object getDetails() {
        return null;
    }

    @Override
    public Object getPrincipal() {
        return admin;
    }

    @Override
    public boolean isAuthenticated() {
        return authenticated;
    }

    @Override
    public void setAuthenticated(boolean isAuthenticated) throws IllegalArgumentException {
        this.authenticated = isAuthenticated;
    }

    @Override
    public String getName() {
        return admin.getUsername();
    }
}

File: src/main/scala/com/tegess/config/security/AdminAuthentication.scala
package com.tegess.config.security

import org.springframework.security.core.{Authentication, GrantedAuthority}
import com.avsystem.commons.jiop.JavaInterop._
import com.tegess.domain.admin.Admin

class AdminAuthentication(val admin: Admin) extends Authentication {
  private var authenticated: Boolean = true

  override def getAuthorities: JCollection[_ <: GrantedAuthority] = {
    admin.getAuthorities
  }

  def getCredentials: Object = {
    admin.getPassword
  }

  def getDetails: Object = {
    null
  }

  def getPrincipal: Object = {
    admin
  }

  def isAuthenticated: Boolean = {
    authenticated
  }

  def setAuthenticated(isAuthenticated: Boolean) {
    this.authenticated = isAuthenticated
  }

  def getName: String = {
    admin.getUsername
  }
}
